model_utils.py

```diff
@@ -5,19 +5,26 @@
 
 
 def prepare_pixelized_dataset(dataset, window_x=5, window_y=5,
-                              y_applied_function=np.asarray):
+                              y_applied_function=np.asarray,
+                              image_by_image=False):
     Xs = []
     ys = []
+    Zs = []
     for datum in dataset:
         img = datum['img']
         mask = datum['mask']
         Xs.append(prepare_pixelized_image(img, window_x, window_y))
+        if image_by_image:
+            Zs.append(img)
         if mask is not None:
             ms = prepare_pixelized_image(mask, window_x, window_y)
-            ys.append((ms.mean(axis=1) > 0.5))
+            nums = np.asarray((ms.mean(axis=1) > 0.5), dtype='int32')
+            ys.append(y_applied_function(nums))
     Xs = np.asarray(Xs)
-    ys = np.squeeze(np.asarray(ys, dtype='int32')).reshape(-1)
-    return Xs.reshape(-1, Xs.shape[-1]), y_applied_function(ys)
+    ys = np.squeeze(ys)
```

Comment:
@mrshu This is due to the fact that applied function is `to_categorical` from `keras.utils.np_utils` https://github.com/fchollet/keras/blob/master/keras/utils/np_utils.py. If you print every element in `ys` you can see that shapes are (3072, 2) or (3072, 1) which results in this error because inside arrays have different shapes. This is caused because function `to_categorical` makes array with  shape `(len(y), nb_classes)` and since number of classes is not provided it assignes to `nb_classes = np.max(y)+1` which for some pictures might be 1 instead of 2. To fix this we should just provide `nb_classes` to the `to_categorical` function.